### ISSUE TITLE
[Opentelemetry] OLTP exporter

### ISSUE THREAD

**Issue by crakjie** [ClickHouse/ClickHouse#55789]
Opentelemetry come with a "common" protocol to export Trace, Message, and Logs to another system. 

It's called `OLTP` for opentelemetry protocol.

I would be nice if we could configure the adress of a service were clickhouse could send his opentelemetry information in the `OLTP` format. 
That would allow clickhouse to export his traces to numbers of service since `Opentelemetry Collector` (which is an `OLTP` service) can receive `OLTP` and convert it to a lot of services proprietary format such _Jager_, _Zipkin_ or _Graphana_.

For a clickhouse user I think he should just have to provide in clickhouse xml configuration file an address and a port for the OLTP service. And probably a configuration to  that allow to disable the feature, to enable but just with a clickhouse user level (queries, sub queries, maybe some background process) and a debbuger level (if needed by clickhouse dev team) that would allow a more granular spanning (query parsing time, query optimisation time ...).

To have a fully monitored clickhouse it could send not only traces but also metric and logs to this service. The configuration of this could be left to the external service, since Opentelemetry collector already have a configuration to filer process and tranform incomming Opentelemetry messages.

**Comment by srikanthccv** [comment]
s/OLTP/OTLP/g

Have you seen this https://clickhouse.com/docs/en/operations/opentelemetry#integration-with-monitoring-systems?

**Comment by crakjie** [comment]
> s/OLTP/OTLP/g
> 
> Have you seen this https://clickhouse.com/docs/en/operations/opentelemetry#integration-with-monitoring-systems?

Yes, but I don't think having a user define exporter is a good solution for something generic like opentelemetry.

**Comment by crakjie** [comment]
It would also be a great way to have a generalized exporter for Clickhouse metrics and logs.

**Comment by ringerc** [comment]
I was _amazed_ to see the use of a query-level push of data to an arbitrary endpoint as the solution for exporting CH tracing data. The lack of support for an OTLP exporter is very surprising.

Are there known major blockers for this? I might see if I can prototype it.
